core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -1626,6 +1642,63 @@ private ExprValue exprToDays(ExprValue date) {
     return new ExprLongValue(date.dateValue().toEpochDay() + DAYS_0000_TO_1970);
   }
 
+  //Helper function to handle a '0000' year for the 'to_seconds' function according to MySQL.
+  private ExprValue handleZeroYear(ExprValue date) {
+    String dateStringVal = date.stringValue();
+
+    //Handle corner case for date of 0000-00-00 according to MySQL docs.
+    if (dateStringVal.contains("0000-00-00")){
+      return ExprNullValue.of();
+    }
+
+    //1970 used as a placeholder to extract correct seconds value.
+    if(dateStringVal.length() > 10) {
+      return(new ExprLongValue(
+          LocalDateTime.parse(date.stringValue().replace("0000-", "1970-"))
+              .toEpochSecond(ZoneOffset.UTC))
+      );
+    }
+
+    return(new ExprLongValue(
+        LocalDate.parse(date.stringValue().replace("0000-", "1970-"))
+            .toEpochSecond(LocalTime.MAX , ZoneOffset.UTC) + 1)
+    );
+  }
+
+  /**
+   * To_seconds implementation for ExprValue.
+   *
+   * @param date ExprValue of Date/Datetime/Timestamp/String type.
+   * @return ExprValue.
+   */
+  private ExprValue exprToSeconds(ExprValue date) {
+
+    //Handle special case of year 0000 according to MySQL docs.
+    if (date.toString().contains("0000-")) {
+      return handleZeroYear(date);
+    }
+
+    return new ExprLongValue(
+        date.datetimeValue().toEpochSecond(ZoneOffset.UTC) + DAYS_0000_TO_1970 * SECONDS_PER_DAY);
+  }
+
+  /**
+   * To_seconds implementation with an integer argument for ExprValue.
+   *
+   * @param date ExprValue of an Integer/Long formatted for a date (e.g., 950501 = 1995-05-01)
+   * @return ExprValue.
+   */
+  private ExprValue exprToSecondsForIntType(ExprValue date) {
+    int day = date.integerValue() % 10;
+    int month = date.integerValue() % 10000 - day;
+    int year = date.integerValue() - month - day;
+
+    LocalDateTime datetime = LocalDateTime.of(1900 + year/10000, month/100, day, 0, 0, 0);
```

Comment:
What if year > 1900?
I think converting datetime types to string in format `yy[yy]mmdd...` and back is already implemented in another function(s). If not - it should be done as a separate function to be re-usable.

Reply:
Addressed in https://github.com/Bit-Quill/opensearch-project-sql/commit/9a2377a5b8ee4bcf5df64a5cea5bf20bd069d32b